### PR TITLE
SHARPENING KITS (but also a new combat knife sprite yoinked from /tg/station)

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
Adds **sharpening kits**. Crafting is as follows:
- craft whetstone whetstone (2 plasteel sheets)
- apply (hit with) 5 sheets of material (preferably not shitty material).
- apply extra material (w_class*2) to the sharpening kit
- apply suitable weapon (not fragile (spears), not broken, etc.)

Also adds the "buckknife" sprite to the kitchen.dmi from tgstation#30680 for the combat knife, because it was being done dirty.
## Why It's Good For The Game
Um.
## Changelog
:cl:
add: Sharpening kits! Hit a whetstone with material sheets to make it, then add sheets and apply a suitable weapon.
/:cl:

### CODE REVIEW

**Comment by VailTheWolf** [comment]
Just make a new knife instead of changing the sprite. Or give it the option to switch sprites.

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 1]
```diff
@@ -8,4 +8,75 @@
...
+
+/obj/item/weapon/material/sharpeningkit/Initialize()
+	. = ..()
+	setrepair()
+
+/obj/item/weapon/material/sharpeningkit/proc/setrepair()
+	repair_amount = material.hardness * 0.1
+	repair_time = material.weight * 0.5
+	sharpen_time = material.weight * 3
+	return
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: trailing return (remove it)

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 2]
```diff
@@ -8,4 +8,75 @@
...
+	thrown_force_divisor = 1
+	var/repair_amount = 5
+	var/repair_time = 40
+	var/sharpen_time = 100
+	var/uses = 0
+
+/obj/item/weapon/material/sharpeningkit/examine(mob/user, distance)
+	. = ..()
+	to_chat(user, "There [uses == 1 ? "is" : "are"] [uses] [material] [uses == 1 ? src.material.sheet_singular_name : src.material.sheet_plural_name] left for use.")
+
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: extra line

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 3]
```diff
@@ -8,4 +8,75 @@
...
+				var/obj/item/SK
+				SK = new /obj/item/weapon/material/sharpeningkit(get_turf(usr), M.material.name)
+				to_chat(usr, "You sharpen and refine \the [src] into \a [SK].")
+				qdel(src)
+				if(SK)
+					usr.put_in_hands(SK)
+		else
+			to_chat(usr, "You need 5 [src] to refine it into a sharpening kit.")
+	return
+
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: extra line

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 4]
```diff
@@ -8,4 +8,75 @@
...
+				M.use(5)
+				var/obj/item/SK
+				SK = new /obj/item/weapon/material/sharpeningkit(get_turf(usr), M.material.name)
+				to_chat(usr, "You sharpen and refine \the [src] into \a [SK].")
+				qdel(src)
+				if(SK)
+					usr.put_in_hands(SK)
+		else
+			to_chat(usr, "You need 5 [src] to refine it into a sharpening kit.")
+	return
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: trailing return

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 5]
```diff
@@ -8,4 +8,75 @@
 	force = 3
 	w_class = ITEMSIZE_SMALL
 	var/repair_amount = 5
-	var/repair_time = 40
\ No newline at end of file
+	var/repair_time = 40
+
+/obj/item/weapon/whetstone/attackby(obj/item/I, mob/usr)
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: do not use usr as a var name, that will collide with byond's built in usr var

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 6]
```diff
@@ -8,4 +8,75 @@
...
+	if(istype(I, /obj/item/stack/material))
+		var/obj/item/stack/material/M = I
+		if(M.amount >= 5)
+			to_chat(usr, "You begin to refine \the [src] with \the [M]...")
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: all the \thes is unnecessary

File: code/game/objects/items/weapons/material/whetstone.dm [hunk 7]
```diff
@@ -8,4 +8,75 @@
...
+		if(uses >= M.w_class*2)
+			if(M.sharpen(src.material.name, sharpen_time, src, usr))
+				uses -= M.w_class*2
+				return
+		else
+			to_chat(usr, "Not enough material to sharpen \the [M]. You need [M.w_class*2] [M.material.sheet_plural_name].")
+			return
+	else
+		to_chat(usr, "You can't sharpen \the [W] with \the [src]!")
+		return
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/whetstone.dm]: trailing return

File: code/game/objects/items/weapons/material/material_weapons.dm [hunk 8]
```diff
@@ -131,7 +134,19 @@
 		to_chat(user, "<span class='warning'>You can't repair \the [src].</span>")
 		return
 
-
+/obj/item/weapon/material/proc/sharpen(var/material, var/sharpen_time, var/kit, mob/living/usr)
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/material_weapons.dm]: usr isn't a good var name

File: code/game/objects/items/weapons/material/material_weapons.dm [hunk 9]
```diff
@@ -27,7 +27,7 @@
 	var/datum/material/material
 	var/drops_debris = 1
 
-/obj/item/weapon/material/New(var/newloc, var/material_key)
+/obj/item/weapon/material/Initialize(var/newloc, var/material_key)
 	..(newloc)
```
Comment (kevinz000) [inline code/game/objects/items/weapons/material/material_weapons.dm]: make sure you know what you're doing when you replace new with initialize

this needs to be . = ..()
don't put in newloc, it's implied

Comment (Hatterhat) [reply code/game/objects/items/weapons/material/material_weapons.dm]: rest assured i did not

File: code/game/objects/items/weapons/material/whetstone.dm (hunk 5 above)
Comment (Hatterhat) [reply code/game/objects/items/weapons/material/whetstone.dm]: how many fucking var names are there that aren't listed

**Pull request merged**